packages/react-devtools-extensions/src/main.js

```diff
@@ -25,6 +25,25 @@ const LOCAL_STORAGE_SUPPORTS_PROFILING_KEY =
 
 const isChrome = getBrowserName() === 'Chrome';
 
+const cachedNetworkEvents = new Map();
+
+// Cache JavaScript resources as the page loads them.
+// This helps avoid unnecessary duplicate requests when hook names are parsed.
+// Responses with a Vary: 'Origin' migt not match future requests.
+// This lets us avoid a possible (expensive) cache miss.
+// For more info see: github.com/facebook/react/pull/22198
+chrome.devtools.network.onRequestFinished.addListener(
+  function onRequestFinished(event) {
+    const {method, url} = event.request;
+    if (method === 'GET' && url.indexOf('.js') > 0) {
+      const {mimeType} = event.response.content;
+      if (mimeType === 'application/x-javascript') {
```

Comment:
There's a few more that should be added here. At least `text/javascript` (which is legacy) and `application/javascript` (which is newer than `application/x-javascript`). There's a few other obscure ones that are probably not worth adding - here's the full list: https://mimesniff.spec.whatwg.org/#javascript-mime-type